src/aind_codeocean_utils/codeocean_job.py

```diff
@@ -0,0 +1,534 @@
+                data_asset_id=results_data_asset_id, everyone="viewer"
+            )
+            logger.info(
+                f"Updating permissions {update_res_perm_response.status_code}"
+            )
+        return reg_result_response
+
+    def run_job(self) -> dict:
+        """
+        Method to process data with a Code Ocean capsule/pipeline.
+        """
+        responses = dict()
+
+        # 1. register data assets (optional)
+        if self.job_config.register_config is not None:
+            logger.info("Registering data asset")
+            register_data_asset_response = (
+                self._register_data_and_update_permissions(
+                    self.job_config.register_config
+                )
+            )
+            data_assets_to_process = deepcopy(
+                self.job_config.run_capsule_config.data_assets
+            )
+            data_assets_to_process.append(
+                ComputationDataAsset(
+                    id=register_data_asset_response.json()["id"],
+                    mount=self.job_config.register_config.mount,
+                )
+            )
+            data_asset_tags = self.job_config.register_config.tags
+            data_asset_custom_metadata = deepcopy(
+                self.job_config.register_config.custom_metadata
+            )
+            responses["register"] = register_data_asset_response
+        else:
+            if len(self.job_config.run_capsule_config.data_assets) == 1:
+                data_asset = self.job_config.run_capsule_config.data_assets[0]
+                data_asset_json = self.co_client.get_data_asset(
+                    data_asset.id
+                ).json()
+                data_asset_name = data_asset_json["name"]
+                data_asset_tags = data_asset_json["tags"]
+                data_asset_custom_metadata = data_asset_json["custom_metadata"]
+            else:
+                # tags and custom_metadata are not propagated if multiple data
+                # assets are provided
+                data_asset_name = None
+                data_asset_tags = []
+                data_asset_custom_metadata = {}
+
+        # 2. run capsule
+        logger.info("Running capsule")
+        run_capsule_response = self._run_capsule(
+            self.job_config.run_capsule_config
+        )
+        responses["run"] = run_capsule_response
+
+        # 3. capture results (optional)
+        if self.job_config.capture_result_config is not None:
+            logger.info("Capturing results")
+            # if the data asset was registered, then the input asset name will
+            # be the same as the new registered data asset name
+            if self.job_config.register_config is not None:
+                input_data_asset_name = (
+                    self.job_config.register_config.asset_name
+                )
+            elif self.job_config.capture_result_config.asset_name is None:
+                assert data_asset_name is not None, (
```

Comment:
data_asset_name isn't defined if `self.job_config.register_config is not None`

Reply:
Let me know if you're updating this or if you want me to

Reply:
Can you tackle it?